server/src/main/java/io/crate/types/BitStringType.java

```diff
@@ -130,6 +131,9 @@ public Streamer<BitString> streamer() {
 
     @Override
     public BitString sanitizeValue(Object value) {
+        if (value instanceof String str) {
+            return new BitString(BitSet.valueOf(str.getBytes(StandardCharsets.UTF_8)), length);
```

Comment:
master version of `DocRefResolver`  uses `implicitCast` which takes length from the value
> return BitString.ofRawBits(str, str.length());

I guess it's fine as we ensure that length would be correct on insert? at least that's how I understood comment
    // implicit cast must not change the length of the value to have proper insert semantics
    // (wrong length = error)